ulx/src/main/java/com/uplink/ulx/drivers/bluetooth/ble/controller/BleDriver.java

```diff
@@ -26,13 +28,16 @@
  * that are common to the two of them, such as the BleDomesticService and the
  * BluetoothManager.
  */
-public class BleDriver extends DriverCommons implements Driver {
+public class BleDriver extends DriverCommons
+        implements Driver,
+        NetworkController.InternetConnectivityDelegate {
 
     private BluetoothManager bluetoothManager;
     private BleDomesticService bleDomesticService;
     private Advertiser advertiser;
     private Browser browser;
     private final SerialOperationsManager operationsManager;
+    private boolean hasInternetConnection;
```

Comment:
It is not obvious if this field is going to be accessed only from a single thread. (Most likely, it won't.) So we need to ensure changes made by the writing thread are visible to the reading ones